diff --git a/test/integration/app.rb b/test/integration/app.rb
index f7e75cc8..5fa1af90 100644
--- a/test/integration/app.rb
+++ b/test/integration/app.rb
@@ -3,6 +3,17 @@ require 'sinatra'
 
 configure do
   set :foo, :bar
+  enable :sessions
+end
+
+helpers do
+  def authenticate!
+    halt 401, 'Unauthorized' unless session[:user_id]
+  end
+end
+
+before '/protected/*' do
+  authenticate!
 end
 
 get '/app_file' do
@@ -10,6 +21,10 @@ get '/app_file' do
   settings.app_file
 end
 
+get '/protected/any_path' do
+  'Protected content'
+end
+
 get '/ping' do
   'pong'
 end
